### PR TITLE
chore(components/e2e) : minor components modification

### PR DESCRIPTION
**Pull request by tlnd-jbrun**
**What is the problem this PR is trying to solve?**
Minor refactos : naming, mutualize WebDriverWait in the parent component,...

**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Component.java [hunk 1]
```diff
@@ -37,20 +43,22 @@ public Component(WebDriver driver, String name, String selector) {
...
-     * @param root Component root element
+     * @param root   Component root element
      */
     public Component(WebDriver driver, String name, WebElement root) {
         LOGGER.info("Component " + name + " from provided root");
         this.driver = driver;
         this.jsExec = (JavascriptExecutor) driver;
         this.name = name;
         this.root = root;
+        this.wait = new WebDriverWait(this.driver, 7);
```
Comment (tlnd-jbrun) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/Component.java]: should be passed as parameter

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Application.java [hunk 2]
```diff
@@ -29,11 +30,36 @@
      * @param driver Selenium WebDriver
      */
     public Application(WebDriver driver) {
+        LOGGER.debug("Construct the Component Application");
```
Comment (ncomont) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/Application.java]: Could you remove this log, please ?

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Application.java [hunk 3]
```diff
@@ -13,6 +13,7 @@
 
     public WebDriver driver;
 
+    // TODO (breaking change) : Rename theses variable in order to respect the Java Convention, and set them private in order to use getter
     public SidePanel SidePanel;
```
Comment (ncomont) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/Application.java]: Why didn't you made this change ?

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Component.java (hunk 1 above)
Comment (ncomont) [reply packages/components/e2e/component-objects/src/main/java/org/talend/component/Component.java]: Why 7 ?

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Component.java [hunk 4]
```diff
@@ -67,17 +75,16 @@ public Component(WebDriver driver, String name, WebElement root) {
      * Get component from its selector
      *
      * @return WebElement found with selector
-     * @throws NotFoundException if no elements are found or if more than one element are found
      */
-    public WebElement getElement() throws NotFoundException {
+    public WebElement getElement() {
```
Comment (ncomont) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/Component.java]: Why did you remove that ?

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/AppHeaderBar.java [hunk 5]
```diff
@@ -32,20 +32,22 @@
 
     /**
      * AppHeaderBar constructor
+     * public constructor in order to be extended in a more specific way for each project
      *
      * @param driver Selenium WebDriver
      */
-    AppHeaderBar(WebDriver driver) {
-        super(driver, NAME, SELECTOR);
+    public AppHeaderBar(WebDriver driver) {
```
Comment (ncomont) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/AppHeaderBar.java]: Should be private

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/AppHeaderBar.java [hunk 6]
```diff
@@ -12,9 +12,9 @@
  */
 public class AppHeaderBar extends Component {
 
-    private static final String NAME = "HeaderBar";
+    private static final String HEADER_BAR_NAME = "HeaderBar";
 
-    private static final String SELECTOR = ".tc-header-bar";
+    private static final String HEADER_BAR_SELECTOR = ".tc-header-bar";
```
Comment (talend-glorieux) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/AppHeaderBar.java]: No need to repeat `HEADER_BAR` since you are already inside the `AppHeaderBar` class.

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/AppHeaderBar.java [hunk 7]
```diff
@@ -12,9 +12,9 @@
  */
 public class AppHeaderBar extends Component {
 
-    private static final String NAME = "HeaderBar";
+    private static final String HEADER_BAR_NAME = "HeaderBar";
```
Comment (talend-glorieux) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/AppHeaderBar.java]: No need to repeat `HEADER_BAR` since you are already inside the `AppHeaderBar` class.

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Component.java [hunk 8]
```diff
@@ -37,20 +43,22 @@ public Component(WebDriver driver, String name, String selector) {
         this.jsExec = (JavascriptExecutor) driver;
         this.name = name;
         this.selector = selector;
+        this.wait = new WebDriverWait(this.driver, 1);
```
Comment (talend-glorieux) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/Component.java]: No wait method should be used. 
It prevents reproducible builds and introduces latency.

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Component.java (hunk 1 above)
Comment (talend-glorieux) [reply packages/components/e2e/component-objects/src/main/java/org/talend/component/Component.java]: No wait method should be used. 
It prevents reproducible builds and introduces latency.

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/list/table/Item.java [hunk 9]
```diff
@@ -17,7 +16,7 @@
  */
 public class Item extends Component {
 
-    private static final String NAME = "Item";
+    private static final String ITEM_NAME = "Item";
```
Comment (talend-glorieux) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/list/table/Item.java]: No need to specify `ITEM`. It's already a private constant of the `Item` class.

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/list/table/Table.java [hunk 10]
```diff
@@ -14,7 +17,7 @@
  */
 public class Table extends Component {
 
-    private static final String NAME = "Table";
+    private static final String TABLE_NAME = "Table";
```
Comment (talend-glorieux) [inline packages/components/e2e/component-objects/src/main/java/org/talend/component/list/table/Table.java]: No need to specify `TABLE`. It's already a private constant of the `Table` class.

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/AppHeaderBar.java (hunk 7 above)
Comment (tlnd-jbrun) [reply packages/components/e2e/component-objects/src/main/java/org/talend/component/AppHeaderBar.java]: i will discard my PR beacause this doen't have any interest but i am not sure that variable shadowing in this case it's a good practice

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Component.java (hunk 1 above)
Comment (tlnd-jbrun) [reply packages/components/e2e/component-objects/src/main/java/org/talend/component/Component.java]: This introduce latency only if you use it, the goal is precisely to the latency. But this was already discussed with Jimmy who thinks that this is not the place to introduce it, but it is better to introduce the wait only when a component needs it. So this PR is useless

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Component.java (hunk 4 above)
Comment (tlnd-jbrun) [reply packages/components/e2e/component-objects/src/main/java/org/talend/component/Component.java]: Because this exception is a RuntimeException so this code is useless

File: packages/components/e2e/component-objects/src/main/java/org/talend/component/Application.java (hunk 3 above)
Comment (tlnd-jbrun) [reply packages/components/e2e/component-objects/src/main/java/org/talend/component/Application.java]: Because the variables names doesn't respect the classic Java checkstyle, and i think it is a bad practice to make them public. We should use getter. But this is a breaking change, and i'am not sure that these reasons are enough significant to deserve the breaking change. So i wanted to open the discussion.
